### PR TITLE
Fixing configuration migration in version upgrade

### PR DESCRIPTION
**Pull request by allenyin96**
Instead of using security_admin tool, use validate and migrate APIs to do the migration.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java [hunk 1]
```diff
@@ -75,7 +77,7 @@
 	protected final ConfigurationRepository cl;
 	protected final ClusterService cs;
 	final ThreadPool threadPool;
-	private String opendistroIndex;
+    protected String opendistroIndex;
```
Comment (hardik-k-shah) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java]: nit: please fix indentation

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java [hunk 2]
```diff
@@ -251,6 +253,11 @@ protected void handleGet(final RestChannel channel, RestRequest request, Client
 		return DynamicConfigFactory.addStatics(loaded);
 	}
 
+    protected final SecurityDynamicConfiguration<?> loadAndDoNotValidate(final CType config, boolean logComplianceEvent) {
```
Comment (hardik-k-shah) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java]: not using this method anywhere..

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java [hunk 3]
```diff
@@ -300,7 +307,7 @@ protected void saveAnUpdateConfigs(final Client client, final RestRequest reques
 		}
 	}
 
-	private static class ConfigUpdatingActionListener<Response> implements ActionListener<Response>{
+    protected static class ConfigUpdatingActionListener<Response> implements ActionListener<Response>{
```
Comment (hardik-k-shah) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java]: nit: indentation

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java (hunk 1 above)
Comment (allenyin96) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java]: Fixed

File: src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java (hunk 3 above)
Comment (allenyin96) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java]: Fixed

**Review by hardik-k-shah — APPROVED** [review]

**Pull request merged**